CI Build job failing on main: sentry_flutter compileReleaseKotlin (JVM target mismatch) — blocks release APKs

The CI Build and Release jobs have been failing on every push to \`main\` since the flutter_lints/package_info_plus/archive dependabot bumps landed. Latest run 24595453936 (sha 6956bc9).

## Failure

\`\`\`
Execution failed for task ':sentry_flutter:compileReleaseKotlin'.
BUILD FAILED in 3m 13s
Gradle task assembleRelease failed with exit code 1
\`\`\`

This is the same "Inconsistent JVM-target compatibility detected for tasks 'compileReleaseJavaWithJavac' (1.8) and 'compileReleaseKotlin' (17)" issue that PRs #45, #48, and #51 tried to fix. The \`allprojects { … plugins.withId("com.android.library") { … compileOptions { targetCompatibility = JavaVersion.VERSION_17 } } }\` block in \`android/build.gradle.kts\` is present on main but not sticking for the \`sentry_flutter\` subproject in CI.

## Why it matters

Every open feature/bugfix PR inherits this failure via rebase, and no new main-built APK can be produced. The user cannot install updated builds.

## Investigation lead

Local build with Flutter 3.41.7 produces a working APK. CI now matches (bumped in PR #51). But CI still fails. Difference may be Android SDK version, Gradle daemon config, or a sentry_flutter version mismatch (we're pulling 8.14.2 — consider bumping to 8.15+ or 9.x which explicitly compiles against JVM 17).

## Fix options to try, in order

1. Bump \`sentry_flutter\` in \`pubspec.yaml\` to a version that ships with JVM 17 bytecode (check 8.16+ or 9.x changelog).
2. If that fails, add explicit \`compileOptions\` + \`kotlinOptions\` overrides in \`android/app/build.gradle.kts\` (not just the root) targeting \`sentry_flutter\` via \`afterEvaluate\` on the library project.
3. If that fails, pin Gradle to the same version archon's local \`./gradlew --version\` uses and make sure AGP matches.

## Acceptance

\`flutter build apk --release\` succeeds in CI on main, and the \`release-apk\` artifact is produced by the Build job.

--- a/android/build.gradle.kts
+++ b/android/build.gradle.kts
@@ -4,18 +4,6 @@ allprojects {
         mavenCentral()
     }
 
-    // Force JVM 17 everywhere — app + every plugin subproject (including
-    // sentry_flutter, which ships with Java 1.8 defaults). Uses reactive
-    // plugins.withId and task-type filters so config applies regardless
-    // of when the target plugin is applied.
-    plugins.withId("com.android.library") {
-        extensions.configure<com.android.build.gradle.LibraryExtension> {
-            compileOptions {
-                sourceCompatibility = JavaVersion.VERSION_17
-                targetCompatibility = JavaVersion.VERSION_17
-            }
-        }
-    }
     plugins.withId("com.android.application") {
         extensions.configure<com.android.build.gradle.AppExtension> {
             compileOptions {
@@ -48,15 +36,14 @@ subprojects {
 subprojects {
     project.evaluationDependsOn(":app")
 
-    // Force JVM 17 on every subproject so plugin-defined Android modules
-    // (e.g. sentry_flutter) don't fall back to Java 1.8 and break the
-    // build with "Inconsistent JVM-target compatibility" on release.
-    plugins.withId("com.android.library") {
-        extensions.configure<com.android.build.gradle.LibraryExtension> {
-            compileOptions {
-                sourceCompatibility = JavaVersion.VERSION_17
-                targetCompatibility = JavaVersion.VERSION_17
-            }
+    // afterEvaluate fires after the subproject's own build script has fully
+    // run, so this override wins over sentry_flutter's own compileOptions
+    // (which resets to Java 1.8). plugins.withId fires mid-evaluation and
+    // gets overridden — that's why PRs #45/#48 didn't stick.
+    afterEvaluate {
+        extensions.findByType<com.android.build.gradle.LibraryExtension>()?.compileOptions {
+            sourceCompatibility = JavaVersion.VERSION_17
+            targetCompatibility = JavaVersion.VERSION_17
         }
     }
     tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {

